packages/utilities/src/utils/validate-subschemas.js

```diff
@@ -27,11 +31,26 @@ const validateNestedSchemas = require('./validate-nested-schemas');
  * (such as those in an 'allOf', 'anyOf' or 'oneOf' property), plus all subschemas
  * of those schemas.
  *
+ * The provided `validate()` function is called with three arguments:
+ * - `subschema`: the composed or nested schema
+ * - `path`: the array of path segments to locate the subschema within the resolved document
+ * - `logicalPath`: the array of path segments to locate an instance of `subschema` within an
+ *    instance of `schema` (the schema for which `validateSubschemas()` was originally called.)
+ *    Within the logical path, an arbitrary array item is represented by `[]` and an arbitrary
+ *    dictionary property is represented by `*`.
+ *
+ * The provided `validate()` function is guaranteed to be called:
+ * - for a schema before any of its composed schemas
+ * - for a schema before any of its nested schemas
```

Comment:
We made need something like this since we may be depending on this behavior.

```suggestion
 * - for a schema before any of its nested schemas
 * - more recently for a schema's parent in the nesting or composition tree than for any of its parents siblings in those trees
```

Reply:
That's a change to make in #709, right? That's one more guarantee that you can add a test for (I believe it should be accurate).

Reply:
Yes.